### ISSUE TITLE
`<Heading/>` - appearance (H1, H2) should be separated from the actual tagName 

### ISSUE THREAD

**Issue by hepiyellow** [wix/wix-style-react#2856]
## The Issue
In several case in WSR components, `<Heading/>` is used, which applies a native tagName of `<h1/>` , `<h2/>`, etc...

Probably most of the time, this is NOT the actual hierarchy of the page content, and may **confuse users using screen-readers**!

### Examples

- Should `<PageHeader/>` title ALWAYS be H1 ?
- Should `<CardAGelleryItem/>` be always H4?
- `<EmptyState/>` - uses H3
- `<TableToolbar.Title/>` - uses H3
- And more...

## Solution

- In some cases, there shouldn't be any `<h?/>` tag, just a simple text. For these cases, we should have the `<Text/>` component be able to set styling which is similar to a heading or change the design.

- I Others, the consumer needs to be able to set the correct tagName.

FYI: @wuwa @milkyfruit @LittleTal

**Comment by shlomitc** [comment]
won't be fixed any time soon - will reopen on request
